bin/specextract

```diff
@@ -828,860 +753,1086 @@ def clip_wmap(wmapfile,threshold,tmpdir):
+        dmgroup.grouptypeval = val
+        dmgroup.ycolumn = "counts"
+        dmgroup.xcolumn = "channel"
+        dmgroup.tabcolumn = ""
+        dmgroup.clobber = clobber
+        dmgroup.verbose = verbose
+
+        dmgroup()
+    
+    return grpout
 
-            #++++++++++++++++++++++++++++++++++++++++++
 
-        except (OSError,IOError):
-            raise IOError("Error generating WMAP w/sky2tdet!")
-			
-        try:
-            # ARF output file
-            arffile = full_outroot + ".arf"
-            
-            # output weight file used in mkrmf
-            weightfile = tempfile.NamedTemporaryFile(suffix=".wfef",dir=tmpdir)
-            
-            mkwarf.punlearn()
 
-            mkwarf.infile = f"{tdetwmap.name}[wmap]"
-            mkwarf.outfile = arffile
-            mkwarf.weightfile = weightfile.name
-            mkwarf.mskfile = mskfile
-            mkwarf.dafile = dafile
-            mkwarf.spectrumfile = ""
-            mkwarf.egridspec = ebin
-            mkwarf.detsubsysmod = "BPMASK=0x03ffff"
-            mkwarf.clobber = "yes"
-            mkwarf.verbose = str(verbose)
+def edit_headers(verbose, infile, key, val, *args, **kwargs):
+    """
+    Update or add the infile header keyword with a specified value.
+    """
+    
+    unit = kwargs.get("unit",None) # optional argument
+    comment = kwargs.get("comment",None)
 
-            mkwarf()
-			
-        except (OSError,IOError):
-            raise IOError("Error generating weighted ARF w/mkwarf!")
-			
-    except IOError as err_msg:
-        print(err_msg)
+    with new_pfiles_environment(ardlib=False,copyuser=False):
+        dmhedit.punlearn()
 
-        raise IOError(f"Failure to create weighted ARF.  Possible causes include: zero counts in the input region; a memory allocation error; or corrupt ARDLIB.  Try running {toolname} with weight=no, binarfmap!=1, or punlearn ardlib.")
+        dmhedit.infile = infile
+        dmhedit.filelist = "none"
+        dmhedit.operation = "add"
+        dmhedit.key = key
+        dmhedit.value = str(val)
+        dmhedit.verbose = str(verbose)
 
-    return arffile, weightfile.name, weightfile, tdetwmap
+        if unit is not None:
+            dmhedit.unit  = str(unit)
 
+        if comment is not None:
+            dmhedit.comment = str(comment)
 
-###############################################################
-# Usage:
-#  check_event_stats(file)
-#
-#
-# Aim:
-#     Use dmlist to determine the event counts
-#     and appropriately error out or proceed with the script
-#
-################################################################
+    return dmhedit()
 
-def check_event_stats(file,refcoord,weights_check=None,ewmap_range_check=None):
 
-    dmlist.punlearn()
 
-    if ewmap_range_check is None:
-        dmlist.infile = file
-    else:
-        dmlist.infile = f"{file}[energy={ewmap_range_check}]"
-        
-    dmlist.opt = "counts"
+def call_calquiz(infile, product, query, file, verbose):
```

Comment:
Ahh, the mysterious `call_calquiz`. Is there any reason why we can't just use the caldb4 module? There could be, as I can't remember how it handles being used in multiple places.